apps/src/templates/instructions/TeacherFeedback.jsx

```diff
@@ -66,36 +137,66 @@ class TeacherFeedback extends Component {
   }
 
   componentDidMount = () => {
-    $.ajax({
-      url: `/api/v1/teacher_feedbacks/get_feedback_from_teacher?student_id=${
-        this.state.studentId
-      }&level_id=${this.props.serverLevelId}&teacher_id=${this.props.teacher}`,
-      method: 'GET',
-      contentType: 'application/json;charset=UTF-8'
-    })
-      .done((data, textStatus, request) => {
+    if (this.props.viewAs === ViewType.Student) {
+      $.ajax({
+        url:
+          '/api/v1/teacher_feedbacks/get_feedbacks?student_id=' +
+          this.props.user +
+          '&level_id=' +
+          this.props.serverLevelId,
+        method: 'GET',
+        contentType: 'application/json;charset=UTF-8'
+      }).done(data => {
         this.setState({
-          latestFeedback: request.status === 204 ? [] : [data],
-          token: request.getResponseHeader('csrf-token'),
-          comment: request.status === 204 ? '' : data.comment
+          latestFeedback: data,
+          comment: data[0].comment,
+          performance: data[0].performance
         });
-      })
-      .fail((jqXhr, status) => {
-        this.setState({errorState: ErrorType.Load});
       });
+    } else if (!this.props.disabledMode) {
+      $.ajax({
+        url: `/api/v1/teacher_feedbacks/get_feedback_from_teacher?student_id=${
+          this.state.studentId
+        }&level_id=${this.props.serverLevelId}&teacher_id=${
+          this.props.teacher
+        }`,
+        method: 'GET',
+        contentType: 'application/json;charset=UTF-8'
+      })
+        .done((data, textStatus, request) => {
+          this.setState({
+            latestFeedback: request.status === 204 ? [] : [data],
+            token: request.getResponseHeader('csrf-token'),
+            comment: request.status === 204 ? '' : data.comment,
+            performance: request.status === 204 ? null : data.performance
+          });
+        })
+        .fail((jqXhr, status) => {
+          this.setState({errorState: ErrorType.Load});
+        });
+    }
   };
 
   onCommentChange = event => {
     this.setState({comment: event.target.value});
   };
 
+  onRubricChange = event => {
+    if (event.target.value === this.state.performance) {
```

Comment:
if our target's value is the same as our performance, why do we want to set performance to null?

Reply:
nevermind! i understand now that i see how `feedbackUnchanged` is calculated on line 230

Reply:
it took me a while to understand this logic too! perhaps it is worth a brief comment explaining.

Reply:
random tip: If you end up making a comment-only change such as this one, you can put `[ci skip]` on the commit message. this will prevent circle from running on it, and (sometimes more importantly) prevent circle from canceling your previous test run.